libsignal-service/src/service_address.rs

```diff
@@ -33,31 +36,93 @@ impl ServiceAddress {
     pub fn pni(&self) -> libsignal_protocol::Pni {
         libsignal_protocol::Pni::from_uuid_bytes(self.uuid.into_bytes())
     }
```

Comment:
Since we now have strong types in this ServiceAddress struct, should we fail the `aci`/`pni` methods when the type is incorrect? We could also rename the methods to `unwrap_aci` and `unwrap_pni`. Currently, we allow "converting" an ACI UUID into a PNI and vice versa, I think.

Reply:
We are currently having a lot of ACI/PNI UUID representations, perhaps we could trim some of them down at some point...

Yes, we should fail the methods if the type is incorrect. Renaming the methods sounds good as well.

Reply:
We could also keep `aci` and make it return an Option instead.